### ISSUE TITLE
Security level authPriv but unencrypted requests are answered

### ISSUE THREAD

**Issue by davidemania** [markabrahams/node-net-snmp#139]
While testing net-snmp to write an agent I noticed a strange behavior, after setting up a user like this

```
const user = {
        name: 'testuser',
        level: snmp.SecurityLevel.authPriv,
        authProtocol: snmp.AuthProtocols.sha,
        authKey: 'pass12345',
        privProtocol: snmp.PrivProtocols.des,
        privKey: 'intotheunknown123'
    }

    authorizer.addUser(user)
```

the authentication and encrypting works well (unauthorized accesses are rejected) if the request uses authPriv, like

accepted-> 
`snmpget_ssl -v3 -l authPriv -utestuser -a sha -Apass12345 -Xintotheunknown123 localhost 1.3.6.1.2.1.1.1.0
`

rejected ->
`snmpget_ssl -v3 -l authPriv -utestuser -a sha -Awrongpassword -Xintotheunknown123 localhost 1.3.6.1.2.1.1.1.0
`

but if I issue an unencrypted request I can get the data even if I should not

accepted-
`snmpget_ssl -v3 -utestuser localhost 1.3.6.1.2.1.1.1.0
`

this of course would be a huge security hole, so I suppose there is something I am doing wrong... but can't figure out what. I tried the provided example and it behaves the same way. Even tried accessing the agent from other hosts, linux and windows but to no avail.

**Comment by markabrahams** [comment]
No you're right there @davidemania - that is a huge security hole unfortunately.  I've added the security level checks now to fix this and published the fix in version 2.10.0 of the npm.  Thanks for reporting this.

**Comment by davidemania** [comment]
Thank you for fixing the issue, I just tested the new version and it looks like it is working correctly.
